client/components/room/Video.jsx

```diff
@@ -27,192 +33,285 @@ class Video extends React.Component {
       mute: 'Mute',
       videoScreen: true,
       startButton: false,
-      pc1: '',
-      pc2: ''
+      isChannelReady: false,
+      isInitiator: false,
+      isStarted: false
     };
+    this.createRoom = this.createRoom.bind(this);
+    this.connectSockets = this.connectSockets.bind(this);
     this.gotStream = this.gotStream.bind(this);
+    this.sendMessage = this.sendMessage.bind(this);
+    this.maybeStart = this.maybeStart.bind(this);
     this.start = this.start.bind(this);
+    this.doCall = this.doCall.bind(this);
+    this.doAnswer = this.doAnswer.bind(this);
+    this.handleCreateOfferError = this.handleCreateOfferError.bind(this);
+    this.setLocalAndSendMessage = this.setLocalAndSendMessage.bind(this);
     this.createPeerConnection = this.createPeerConnection.bind(this);
     this.handleIceCandidate = this.handleIceCandidate.bind(this);
-    this.getName = this.getName.bind(this);
-    this.getOtherPc = this.getOtherPc.bind(this);
-    this.onCreateOfferSuccess = this.onCreateOfferSuccess.bind(this);
-    this.onCreateAnswerSuccess = this.onCreateAnswerSuccess.bind(this);
-    this.onSetLocalSuccess = this.onSetLocalSuccess.bind(this);
-    this.onSetRemoteSuccess = this.onSetRemoteSuccess.bind(this);
     this.onSetSessionDescriptionError = this.onSetSessionDescriptionError.bind(this);
-    this.gotRemoteStream = this.gotRemoteStream.bind(this);
-    this.handleVideoScreen = this.handleVideoScreen.bind(this);
-    this.handleAudio = this.handleAudio.bind(this);
+    // this.requestTurn = this.requestTurn.bind(this);
+    this.handleRemoteStreamAdded = this.handleRemoteStreamAdded.bind(this);
+    this.handleRemoteStreamRemoved = this.handleRemoteStreamRemoved.bind(this);
+    this.hangup = this.hangup.bind(this);
+    this.handleRemoteHangup = this.handleRemoteHangup.bind(this);
+    this.stop = this.stop.bind(this);
+    // this.handleVideoScreen = this.handleVideoScreen.bind(this);
+    // this.handleAudio = this.handleAudio.bind(this);
   }
 
   componentDidMount() {
-    socket.emit('Hello', 'Hello');
-    const localVideo = document.getElementById('localVideo');
-    const remoteVideo = document.getElementById('remoteVideo');
+    let localVideo = document.getElementById('localVideo');
+    let remoteVideo = document.getElementById('remoteVideo');
```

Comment:
While not MVP, I think there might be a way to set src on these elements without using getElementById.